doc/marlowe/tutorials/marlowe-data.rst

```diff
@@ -42,56 +46,56 @@ In modelling basic parts of Marlowe we use a combination of Haskell
-the notion of ownership of a contract role, and make it tradable. So you
-can sell your loan or buy a share of a role in some contract.
+``Role`` parties will look like ``Role "alice"``, ``Role "bob"``
+and so on. However, Haskell allows us to present and read in these
+values more concisely (by declaring a custom instance of ``Show`` and
+using *overloaded strings*) so that these can be input and read as used above,
+``"alice"``, ``"bob"`` etc.
 
-::
+A Marlowe *account* holds amounts of multiple currencies and/or fungible
+and non-fungible tokens. A concrete amount is indexed by a ``Token``,
+which is a pair of a *currency symbol* and a *token name*, both given by a ``ByteString``. 
 
-   tradeRoles = CurrencySymbol "TradeRoles"
-   buyer = TokenName "buyer"
-   seller = TokenName "seller"
-   account = fromList[(Token tradeRoles buyer, 1), (ada 1000)]
+.. code:: haskell
 
-Here ``account`` holds a ``buyer`` token of currency "TradeRoles", and
-1000 lovelace.
+   data Token = Token ByteString ByteString
 
-A ``Party`` is represented as either a public key hash or a role name.
+   Cardano’s Ada token is 
 
-.. code:: haskell
+   .. code:: haskell
+   
+      ada = Token adaSymbol adaToken
+   
+But you are free to create your own currencies and tokens using the native token facility of Cardano.   
+You can think
+of an Account as used above, a map from ``Token`` to ``Integer`` and so all the accounts in a contracts can be modelled like this:
 
-   data Party  = PK PubKeyHash | Role TokenName
+.. code:: haskell
 
-In order to progress a Marlowe contract, a party must provide an
-evidence. For ``PK`` party that would be a valid signature of a
-transaction signed by a private key of a public key that hashes to
-party’s ``PubKeyHash``, similarly to Bitcoin’s
-``Pay to Public Key Hash`` mechanism. For a ``Role`` party the evidence
-is spending a ``role token`` within the same transaction, usually to the
-same owner.
+   type Accounts = Map (AccountId, Token) Integer
 
-So, ``Role`` parties will look like ``(Role "alice")``, ``(Role "bob")``
-and so on. However, Haskell allows us to present and read in these
-values more concisely (by declaring a custom instance of ``Show`` and
-using *overloaded strings*) so that these can be input and read as
-``"alice"``, ``"bob"`` etc.
+Tokens of a currency can represent roles in a contract, e.g ``"buyer"`` and
+``"seller"``. Think of a legal contract in the sense of "hereafter referred
+to as used above, the Performer/Vendor/Artist/Consultant". This way we can decouple
+the notion of ownership of a contract role, and make it tradable. So you
+can sell your loan or buy a share of a role in some contract.
 
-Slot numbers and amounts of money are treated in a similar way; with the
-same show/overload approach they will appear in contracts as numbers:
+Slot numbers and amounts of money are treated in a similar way: with the
+same show/overload approach as used above, they will appear in contracts  numbers:
```

Comment:
I think the `as` before numbers is still necessary, also the two `:` in a row are a bit strange, maybe `;` was better